magento/magento2
# 2.2 develop 11032 Unable to add new options to swatch attribute

ProductAttributeOptionManagementInterface#add() doesn't do anything.

### Description

On the beforeBeforeSave(Attribute $attribute) in the case of a swatch attribute, expects an array with key swatchtext / swatchvisual ( first in case of textswatch, second in case of visualswatch) that overwrites the option array originally given, so if this is empty, it will be emptied as well. So I added a plugin that right before doing this, creates the expected array that will be used later for the creation of the actual option. 

### Fixed Issues (if relevant)

1. magento/magento2#11032: Unable to add new options to swatch attribute
2. magento/magento2#9410: Impossible to add swatch options via Service Contracts if there is no existing swatch option for attribute

### Manual testing scenarios

1.  Call the rest api http://<magento-url>/index.php/rest/V1/products/attributes/color/options to add a new color option with the following input 
`{
  "option": {
    "label": "new color",
    "value": "",
    "sort_order": 0,
    "is_default": true,
    "store_labels": [
      {
        "store_id": 0,
        "label": "new color"
      },
      {
        "store_id": 1,
        "label": "new color"
      }
    ]
  }
}`
2. Check from the admin area that your new option is there or call the rest api http://<magento-url>/index.php/rest/V1/products/attributes/color/options

### Contribution checklist
